packages/javascript-tracker/test/integration.test.ts

```diff
@@ -5,10 +5,15 @@ import {
   trackSearch,
   trackSearchClick,
 } from "@elastic/behavioral-analytics-javascript-tracker";
-import mock from "xhr-mock";
+import { Tracker } from "@elastic/behavioral-analytics-tracker-core";
```

Comment:
Would still like to keep this integration test wide, validating how fetch is being called, validating things like authorisation, the session & user attributes have been added by core etc. Its more of a contract test between the core and the javascript-tracker. Could we instead mock fetch and validate fetch is being call with?